src/platform/silabs/wifi/icd/WifiSleepManager.cpp

```diff
@@ -0,0 +1,170 @@
+ */
+CHIP_ERROR ConfigureDeepSleep()
+{
+    ReturnLogErrorOnFailure(ConfigurePowerSave(RSI_SLEEP_MODE_8, DEEP_SLEEP_WITH_RAM_RETENTION, 0));
+    return CHIP_NO_ERROR;
+}
+
+/**
+ * @brief Configures the Wi-Fi chip to go to High Performance.
+ *        Function doesn't change the broad cast filter configuration.
+ *
+ * @return CHIP_ERROR CHIP_NO_ERROR if the configuration of the Wi-Fi chip was successful; otherwise CHIP_ERROR_INTERNAL
+ */
+CHIP_ERROR ConfigureHighPerformance()
+{
+    ReturnLogErrorOnFailure(ConfigurePowerSave(RSI_ACTIVE, HIGH_PERFORMANCE, 0));
+    return CHIP_NO_ERROR;
+}
+#endif // SLI_SI917
+
+} // namespace
+
+namespace chip {
+namespace DeviceLayer {
+namespace Silabs {
+
+// Initialize the static instance
+WifiSleepManager WifiSleepManager::mInstance;
+
+CHIP_ERROR WifiSleepManager::Init()
+{
+    return VerifyAndTransitionToLowPowerMode(PowerEvent::kGenericEvent);
+}
+
+CHIP_ERROR WifiSleepManager::RequestHighPerformance()
+{
+    VerifyOrReturnError(mHighPerformanceRequestCounter < std::numeric_limits<uint8_t>::max(), CHIP_ERROR_INTERNAL,
+                        ChipLogError(DeviceLayer, "High performance request counter overflow"));
+
+    mHighPerformanceRequestCounter++;
+
+    // We don't do the mHighPerformanceRequestCounter check here; the check is in VerifyAndTransitionToLowPowerMode function
+    ReturnErrorOnFailure(VerifyAndTransitionToLowPowerMode(PowerEvent::kGenericEvent));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WifiSleepManager::RemoveHighPerformanceRequest()
+{
+    VerifyOrReturnError(mHighPerformanceRequestCounter > 0, CHIP_NO_ERROR,
+                        ChipLogError(DeviceLayer, "Wi-Fi configuration already in low power mode"));
+
+    mHighPerformanceRequestCounter--;
+
+    // We don't do the mHighPerformanceRequestCounter check here; the check is in VerifyAndTransitionToLowPowerMode function
+    ReturnErrorOnFailure(VerifyAndTransitionToLowPowerMode(PowerEvent::kGenericEvent));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WifiSleepManager::HandlePowerEvent(PowerEvent event)
+{
+    switch (event)
+    {
+    case PowerEvent::kCommissioningComplete:
+        ChipLogProgress(AppServer, "WifiSleepManager: Handling Commissioning Complete Event");
+        mIsCommissioningInProgress = false;
+
+        // TODO: Remove High Performance Req during commissioning when sleep issues are resolved
+        WifiSleepManager::GetInstance().RemoveHighPerformanceRequest();
+        break;
+
+    case PowerEvent::kConnectivityChange:
+    case PowerEvent::kGenericEvent:
+        // No additional processing needed for these events at the moment
+        break;
+
+    default:
+        ChipLogError(AppServer, "Unknown Power Event");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WifiSleepManager::VerifyAndTransitionToLowPowerMode(PowerEvent event)
```

Comment:
Most functions perform verifications, I don't see a reason to explicitly name this one VeryAnd... Thad would make sense in a set of other VeryAnd*, or *TransitionToLowPowerMode, but this doesn't seem to be the case.
```suggestion
CHIP_ERROR WifiSleepManager::TransitionToLowPowerMode(PowerEvent event)
```

Reply:
Since the function contains all the buisness logic to determine the lowest power mode, I think Verify* has an additionnal value to the clarity of the function.